### PR TITLE
Dc 4501 - Update '/messages' end point with the code from message back-end service

### PR DESCRIPTION
**Pull request by satyajosyula2020**

### CODE REVIEW

File: app/uk/gov/hmrc/securemessage/models/core/MessageFilter.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+package uk.gov.hmrc.securemessage.models.core
+
+import play.api.libs.json.{ Json, Reads }
+import uk.gov.hmrc.common.message.model.Regime
+
+case class MessageFilter(
+  taxIdentifiers: Seq[String] = List(),
+  regimes: Seq[Regime.Value] = List(),
+  countOnly: Boolean = false)
```
Comment (PaulThor) [inline app/uk/gov/hmrc/securemessage/models/core/MessageFilter.scala:25]: We should remove this count, as there is a separate Endpoint for this

File: app/uk/gov/hmrc/securemessage/controllers/binders/package.scala [hunk 2]
```diff
@@ -48,4 +50,28 @@ package object binders {
...
+
+  implicit def messageFilterBinder(
+    implicit seqBinder: QueryStringBindable[Seq[String]],
+    boolBinder: QueryStringBindable[Boolean]): QueryStringBindable[MessageFilter] =
+    new QueryStringBindable[MessageFilter] {
+      override def bind(key: String, params: Map[String, Seq[String]]): Option[Either[String, MessageFilter]] =
+        for {
+          taxIdentifiers <- seqBinder.bind("taxIdentifiers", params).orElse(Some(Right(List[String]())))
+          regimes        <- seqBinder.bind("regimes", params).orElse(Some(Right(List[String]())))
+          countOnly      <- boolBinder.bind("countOnly", params).orElse(Some(Right(false)))
```
Comment (PaulThor) [inline app/uk/gov/hmrc/securemessage/controllers/binders/package.scala:62]: We don't need count

File: app/uk/gov/hmrc/securemessage/controllers/model/MessagesResponse.scala [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+import play.api.libs.functional.syntax._
+import play.api.libs.json._
+import uk.gov.hmrc.common.message.model.MessagesCount
+import uk.gov.hmrc.http.controllers.RestFormats
+import uk.gov.hmrc.mongo.play.json.formats.MongoFormats
+import uk.gov.hmrc.securemessage.models.core.{ Letter, RecipientName }
+
+import scala.annotation.tailrec
+
+final case class MessageListItem(
```
Comment (PaulThor) [inline app/uk/gov/hmrc/securemessage/controllers/model/MessagesResponse.scala:30]: We should re-use the same class as before

**Review by PaulThor — COMMENTED** [review]
Great work, just a couple of comments pls review

File: app/uk/gov/hmrc/securemessage/models/core/MessageFilter.scala (hunk 1 above)
Comment (satyajosyula2020) [reply app/uk/gov/hmrc/securemessage/models/core/MessageFilter.scala:25]: Good catch, overlooked by me, thanks

**Review by PaulThor — APPROVED** [review]

**Pull request merged**